[21606] Add AML Dashboard tutorials (#16)
eProsima-Private/AML-Tools: docs/rst/amldashboard/user_manual/interfaces.rst

.. include:: /rst/exports/alias.include
.. include:: /rst/exports/roles.include

.. _amldashboard_interfaces:

##########

> Interfaces

Reviewer: Go to the comments made in the PR https://github.com/eProsima-Private/AML-Tools/pull/12 .

Please add pictures of the display of each tab.

##########

The *AML Dashboard* integrates six main interfaces:

Data Collection
===============

> The first screen comprises five distinct areas that enable users to swiftly create their own data sets for training an |aml| model.

Reviewer: ```suggestion
The first tab comprises five distinct areas that enable users to swiftly create their own data sets for training an |aml| model.
```

1. **Exploring Data Input**: The *AML Dashboard* collection interface is designed to help non-computer scientists easily create and understand data collection.
   It features two video representations:

   - **Ground Truth Video Stream**: Shows the actual video feed.

   - **Overlaid Skeleton Video**: Displays the skeleton recognized by the hand recognition algorithm.

   These visual aids allow users to test and explore the aspects captured by the model.

2. **Collecting Data**: Users can create their own data sets to train an |aml| model.

>    A drop-down menu allows the selection of one out of four musical instruments for demonstration purposes.

Reviewer: This is not implemented yet. Please make sure the tutorial matches the real demo.

> 3. **Data Set Overview**: Each collected image is displayed in a data set overview, where users can explore, move, or delete images in the training stack.

Reviewer: ```suggestion
3. **Data Set Overview**: Each collected image is displayed in a data set browser, where users can explore, move, or delete images in the training stack, as well as managing the labels of the training set.
```

   This feature is crucial for iterative training, enabling users to remove or add new labeled data to improve the |aml| algorithm as needed.

.. _amldashboard_interfaces_training:

Training
========

> The training page is separate from the data collection page to educate users on the various stages of the :term:`ML` life-cycle.

Reviewer: ```suggestion
The training tab is separated from the Data Management tab to educate users on the various stages of the :term:`ML` life-cycle.
```

It is divided into two sections:

1. **AML Model Training Information (Left Side)**:

>    - **Training Information**: Users can set parameters such as the number of iterations.

Reviewer: ```suggestion
   - **Training Information**: Users can set parameters such as the number of iterations and percentage of data allocated to each execution.
```

>    - **Training Visualization**: Displays the current state of training.

Reviewer: ```suggestion
   - **Training Status**: Displays the current state of training.
```

2. **Comparison Model Training (Right Side)**:

   - **Model Parameters**: Users can set or alter parameters for a comparison model, such as a Multi-layer Perceptron (MLP) classifier.
     This model can be replaced with any other type of classifier for direct comparison with |aml|.

   - **Training Progress**: A loading bar shows the projected training duration.
     Below it, variations in loss and accuracy per epoch are visualized, helping experts assess the classification quality and data noise level.

.. _amldashboard_interfaces_fetching:

Fetching
========

> Retrieve trained |aml| statistics and Models effortlessly.

Reviewer: ```suggestion
Its purpose is to retrieve trained |aml| statistics and Models effortlessly.
```

It includes options to search for |aml| statistics and request an |aml| model.

It is divided into the following sections:

1. **Status Display**: Shows the current status of collaborative learning, statistics, and model fetching.

2. **AML Statistics Fetcher**:

   - **Search Button**: Search for |aml| statistics.

   - **Statistics Display**: Display the fetched statistics

3. **AML Model Fetcher**:

   - **Request Button**: Request an |aml| model.

   - **Model Display**: Display the fetched model.

> Explore Models

Reviewer: Please make sure the names in the tutorial match the names in the demo.
```suggestion
Batch Prediction. 
```

==============

The model exploration interface provides space for developers to add any desired model comparison matrix.
The Marcelle framework supports various comparison modules and allows users to create custom :term:`GUI` elements for specific calculations.

In the current setup, a confusion matrix is provided based on the trained model.
Users must actively trigger the update of the matrix, facilitating a clear before-and-after comparison.
Future work aims to extend this section with more targeted visualizations utilizing the |aml| output structure.

> Music Exploration

Reviewer: Please make sure the names in the tutorial match the real names in the demo
```suggestion
Real-Time Predictions
```

=================

> After training the models, users can move to the music exploration tab.

Reviewer: ```suggestion
After training the models, users can move to the Real-time Prediction tab.
```

This part of *AML Dashboard* allows users to explore music patterns using the self-trained models.

Key features include:

1. **Model Selection**: Users can toggle between |aml| and *NN* models for real-time predictions.

2. **Camera Activation**: Starting the camera initiates real-time predictions from the activated model.

3. **Real-Time Predictions**: Predictions are shown in real-time.
   If both models run, the prediction confidence of the |aml| model determines the sound pattern volume.

Sound is produced by looping audio tracks of musical instruments assigned during data collection.
Users can recreate gestures to explore sound intersections using *AML Dashboard*.

Context Broker interaction
==========================

> Create and update data in the Context Broker and receive solutions (inferences) from the |aml| model.

Reviewer: ```suggestion
This tab's purpose is to create and update data in the Context Broker and receive solutions (inferences) from the |aml| model.
```

1. **Fiware Node**:

   - **Parameters**: Specify the Fiware Node parameters and Context Broker entity ID and attributes.

   - **Create Button**: Create the node using the provided parameters.

   - **Status Update**: Shows whether the node has been created successfully.

2. **Context Broker Data**:

   - **Data Upload**: Drag and drop an image or upload a file in the designated area to upload data.

   - **Post Data Button**: Send the data to the Context Broker.

>    - **Data Status**: Indicates whether the data has been successfully posted.

Reviewer: ```suggestion
   - **Data Status**: Indicates whether the data has been successfully loaded and posted.
```

>     :align: center
>     :width: 100

Reviewer: ```suggestion
    .. figure:: /rst/figures/amldashboard/aml-dashboard_data_management.png

```

Reviewer: This is because when the documentation compiles, the image appears too small.

    :align: center

>     :width: 100

Reviewer: ```suggestion
    .. figure:: /rst/figures/amldashboard/aml-dashboard_trained.png
```

Reviewer: Same for all figures in this file
